spec/support/authentication_helpers.rb

```diff
@@ -59,4 +49,29 @@ def password_from_config_or_prompt
       password.strip
     end
   end
+
+  def ensure_username!
+    self.username ||= username_from_config_or_prompt
+  end
+
+  def ensure_password!
+    self.password ||= password_from_config_or_prompt
+  end
+
+  def submit_credentials_if_needed
+    return unless page.has_text?('SUNet ID', wait: Settings.post_authentication_text_timeout)
+
+    ensure_password!
```

Comment:
as far as i could tell, password (unlike sunet) is only used for login, so it seemed fine to keep it coupled to login form submission.  happy to adjust as needed if testing by others indicates that this doesn't work (this is the main reason i'd like someone who doesn't use cardinal key to test).